src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php

```diff
@@ -169,14 +188,14 @@ protected function getMappingDriverBundleConfigDefaults(array $bundleConfig, \Re
             if (\in_array($bundleConfig['type'], ['annotation', 'staticphp', 'attribute'])) {
                 $bundleConfig['dir'] = $bundleDir.'/'.$this->getMappingObjectDefaultName();
             } else {
-                $bundleConfig['dir'] = $bundleDir.'/'.$this->getMappingResourceConfigDirectory();
+                $bundleConfig['dir'] = $bundleDir.'/'.$this->getMappingResourceConfigDirectory($bundleDir);
             }
         } else {
             $bundleConfig['dir'] = $bundleDir.'/'.$bundleConfig['dir'];
         }
 
         if (!$bundleConfig['prefix']) {
-            $bundleConfig['prefix'] = $bundle->getNamespaceName().'\\'.$this->getMappingObjectDefaultName();
+            $bundleConfig['prefix'] = $bundleMetadata['namespace'].'\\'.$this->getMappingObjectDefaultName();
```

Comment:
can really this be different? because if not, we'd better add `$bundleDir` as last argument
or we could extend ReflectionClass and return the correct value from getNamespaceName?
I'm looking for a way to get rid of the deprecation if possible.

Reply:
My thinking was that the information that is actually needed is already collected in `kernel.bundles_metadata` and there isn't a need to reflect the Bundle class at all, which is how I landed on the implementation in this PR right now.  It's harder to deal with from a B/C perspective to fully remove the reflected class, but it makes it less error-prone IMO if there's only one data source (the bundle metadata).

Adding `$bundleDir` as an argument would definitely work though.

Reply:
I understand, but let's do it the most seamless way, aka with the extra argument IMHO.

Reply:
PR updated